server/routes/nl/helpers.py

```diff
@@ -404,10 +404,9 @@ def abort(error_message: str,
           blocked: bool = False,
           test: str = '',
           client: str = '') -> Dict:
-  query = str(escape(shared_utils.remove_punctuations(original_query)))
-  escaped_context_history = []
-  for ch in context_history:
-    escaped_context_history.append(escape(ch))
+  query = shared_utils.escape_strings(
+      shared_utils.remove_punctuations(original_query))
+  escaped_context_history = shared_utils.escape_strings(context_history)
```

Comment:
What is the motivation for this change?  Might be better to make changes like in a separate PR (OK for this one)?

Also, are there other places we escape context_history?

Reply:
this is because when we just used `escape(ch)`, it turned the context_history into a list of strings instead of a list of dicts, which then caused problems when the context was being loaded for the next query because load_utterance is expecting a dict, not a string. This came up because I added a test case where there is a query that goes down this abort path and then the query after that would error out.

Reply:
Hmm, I see, the flow of context in the abort case might be untested/unknown.  What happens with the stats flow?  Might be worth following up...

Reply:
If you agree, have a TODO in abort maybe, for follow up.

Reply:
added TODO